### PR TITLE
Lib updates, fixes for #199, implement #160

### PR DESCRIPTION
**Pull request by jeffdc**

### CODE REVIEW

**Comment by jeffdc** [comment]
For #206 added stricter validation for species names on the admin forms for gall and host. 

These are all good:
```
Foo bar
Foo bar-baz
Foo bar-baz-boo
Foo bar-baz (boo)
Foo bar-baz (boo) (boo)
Foo x bar
Foo x bar-baz
Foo x bar-baz (boo)
Foo x bar-baz (boo) (boo)
```

These are not:
```
Foo s bar
Foo bar agamic
foo bar
F bar
 Foo bar
```

File: migrations/016-gallformers.sql [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+CREATE TABLE gallform (
+    id       INTEGER PRIMARY KEY NOT NULL,
+    gall_id  INTEGER,
+    form_id INTEGER,
+    FOREIGN KEY (gall_id) REFERENCES gall (id) ON DELETE CASCADE,
+    FOREIGN KEY (form_id) REFERENCES form (id) ON DELETE CASCADE
+);
+
+PRAGMA foreign_keys=ON;
+
```
Comment (jeffdc) [inline migrations/016-gallformers.sql:50]:
```suggestion
INSERT INTO form (id, form, description) VALUES (NULL, 'gall', 'Definition pending.');
```

Comment (jeffdc) [reply migrations/016-gallformers.sql:50]: @Megachile We will add gall. Just need a definition.

**Comment by Megachile** [comment]
Add Form:Gall by default to everything except: 
Albuginaceae
Buprestidae
Cerambycidae
Cerococcidae
Cicadellidae
Closteroviridae
Coccidae
Diaspididae
Erysiphaceae
Kermesidae
Microstromataceae
Pucciniastraceae
Rhytismataceae
Sesiidae
Steingeliidae
Tombusviridae
Tortricidae

File: migrations/016-gallformers.sql [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+UPDATE texture SET description = 'Multiple colors on the surface of the gall mix irregularly.' WHERE texture = 'mottled';
+UPDATE texture SET description = 'The surface of the gall is covered in dots, often red, that secrete sticky resin.' WHERE texture = 'resinous dots';
+UPDATE texture SET description = 'The gall is covered in sharp spines, prickles, etc.' WHERE texture = 'spiky/thorny';
+UPDATE texture SET description = 'The walls of the gall (when fresh) are juicy if cut.' WHERE texture = 'succulent';
+UPDATE texture SET description = 'Covered in a whitish layer of fine powder or wax that can be easily rubbed off.' WHERE texture = 'glaucous';
+UPDATE texture SET description = 'Galls releasing sugary solution. Often visible as a shiny wetness, but can be more apparent in the ants and wasps it attracts.', texture = 'honeydew' WHERE texture = 'sticky'; 
+UPDATE texture SET description = 'The gall is hard and incompressable to the touch, generally because they are woody, thick-walled, but sometimes with an almost plastic-like texture.' WHERE texture = 'stiff';
+-- remap felt (1) to pubsecent (2)
```
Comment (Megachile) [inline migrations/016-gallformers.sql:101]: pubescent misspelled--not sure if this matters

File: libs/utils/util.ts [hunk 3]
```diff
@@ -163,3 +167,5 @@ export const extractGenus = (n: string): string => {
 };
 
 export const sessionUserOrUnknown = (user: string | null | undefined): string => (user ? user : 'UNKNOWN!');
+
+export const SPECIES_NAME_REGEX = /(^[A-Z][a-z]+ (x|X)?\s?[a-z-]+\s?(\(.+\)\s*)*$)/;
```
Comment (jeffdc) [inline libs/utils/util.ts:171]: Add comment explaining the valid formats

File: libs/db/gall.ts [hunk 4]
```diff
@@ -580,6 +588,26 @@ export const getCells = (): TaskEither<Error, CellsApi[]> => {
...
 };
 
+const adaptForm = (form: form[]): FormApi[] =>
+    form.map((c) => ({
+        ...c,
+        description: O.fromNullable(c.description),
+    }));
+
+/**
+ * Fetches all gall cells
```
Comment (jeffdc) [inline libs/db/gall.ts:598]: Wrong comment.

File: libs/db/species.ts [hunk 5]
```diff
@@ -106,40 +106,53 @@ export const speciesUpdateData = (sp: SpeciesUpsertFields) => ({
...
+                    },
                 },
             },
         },
-    },
-});
+    });
+
+    // handle section for hosts
+    // This is a hack to deal with the fact that a Secion may not be presen. Prisma craps out if you pass an empty element
```
Comment (jeffdc) [inline libs/db/species.ts:133]: Typos

File: prisma/schema.prisma [hunk 6]
```diff
@@ -55,6 +55,13 @@ model color {
   gallcolor gallcolor[]
 }
 
+model form {
+  id          Int         @id @default(autoincrement())
```
Comment (jeffdc) [inline prisma/schema.prisma:59]: remove

Comment (jeffdc) [reply prisma/schema.prisma:59]: never mind

**Pull request merged**